lighthouse-core/config/config.js

```diff
@@ -69,7 +69,7 @@ function assertValidPasses(passes, audits) {
     pass.gatherers.forEach(gathererDefn => {
       const gatherer = gathererDefn.instance;
       foundGatherers.add(gatherer.name);
-      const isGatherRequiredByAudits = requiredGatherers.has(gatherer.name);
```

Comment:
should we warn in the scenario that an optional artifact is missing? I think that depends on our imagined long-term use case and how close to "required" these optionalArtifacts are supposed to be

Reply:
I don't like warnings that are "maybe this is a warning, IDK, you figure it out!" :)

Reply:
well in the case where we're trying to pull it in for them but can't that seems like prime warning territory it's basically just a notch below required and a notch below complete failure is a warning.

if it's the scenario where they're truly optional and we don't try to pull the gatherer in for them, then I agree sounds like no warning is appropriate.

IMO it's more egregious at the moment that the default CLI run results in warnings for unused artifacts that are only there for potential plugin usage.

Reply:
since it sounds like we're going with the former, but internal only, I'm fine with no warning.